pkg/cmd/roachtest/tests/ptp.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	spec2 "github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func registerPTP(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:  "ptp",
+		Owner: registry.OwnerKV,
+		// Don't reuse this cluster after this test, since we're installing kernel modules etc.
+		// In practice I don't see how this could cause problems but it's not worth any risk.
+		Cluster: r.MakeClusterSpec(3, spec2.ReuseNone()),
+		Leases:  registry.MetamorphicLeases,
+		// We're unlikely to break PTP support on the regular and this test does not meaningfully
+		// exercise anything else.
+		Suites: registry.Suites(registry.Weekly),
```

Comment:
Do you want to have this run nightly for a while to get some miles on it? Then we can drop to weekly? No strong opinion.